src/EditorFeatures/DiagnosticsTestUtilities/Diagnostics/AbstractUserDiagnosticTest.cs

```diff
@@ -102,24 +103,21 @@ internal override Task<CodeRefactoring> GetCodeRefactoringAsync(EditorTestWorksp
 
         protected static void AddAnalyzerToWorkspace(Workspace workspace, DiagnosticAnalyzer analyzer, TestParameters parameters)
         {
-            AnalyzerReference[] analyzeReferences;
+            AnalyzerReference[] analyzerReferences;
             if (analyzer != null)
             {
-                Contract.ThrowIfTrue(parameters.testHost == TestHost.OutOfProcess, $"Out-of-proc testing is not supported since {analyzer} can't be serialized.");
```

Comment:
instead of asserting and not allowing this.  We just make it so AnalyzerImageReference do work in OOP when in test scenarios.